src/Microsoft.PowerShell.Commands.Utility/commands/utility/select-object.cs

```diff
@@ -337,6 +337,11 @@ private void ProcessExpressionParameter()
             if (ExcludeProperty != null)
             {
                 _exclusionFilter = new MshExpressionFilter(ExcludeProperty);
+                if ((Property == null) || (Property.Length == 0))
+                {
```

Comment:
1.The current tests do not catch it. I think we should add this in tests.

2.There is no difference between:
```powershell
 $s = Get-Process -Id $pid | Select-Object -ExpandProperty Modules
$s[0] | fl *
```
and
```powershell
$s = Get-Process -Id $pid | Select-Object -ExcludeProperty FileName -ExpandProperty Modules
$s[0] | fl *
```
i.e.  ExpandProperty skip ExcludeProperty. So the change could be corrected without  breaking.

3.The cmdlet is very intuitive to the user UX and very controversial:
ExcludeProperty is skipped w/o Property
ExpandProperty skip ExcludeProperty
ExpandProperty with Property throw exception
Compare:
```powershell
1 | Select-Object
"1" | Select-Object
1 | Select-Object -Property *
"1" | Select-Object -Property *
```
and many more.
And it is a tricky code, and any even slightly change this code can easily break down. 

(In general I think this cmdlet should be replaced with a newer, which meets modern requirements to cmdlets.)